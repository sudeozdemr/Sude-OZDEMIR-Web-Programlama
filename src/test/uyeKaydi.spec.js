import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'
import register from '../components/uyeKaydi.vue'


const localVue = createLocalVue();
localVue.use(Vuex);
describe('uyeOl', () => {
    let wrapper;
    const mockFunction = jest.fn();
    describe('Kaydol Butonuna Basıldığında', () => {
        beforeEach(() => {
            wrapper = mount(register, {
                localVue,
                data() {
                    return {
                        userInfo: {
                            email: 'test@test.com',
                            password: '111222',
                            name: "Ali",
                            surname: "Kaya"
                        }
                    };
                },
                store: new Vuex.Store({
                    actions: {
                        register: mockFunction,
                    },
                    state: {
                        cart: []
                    }
                }),
                mocks: {
                    $router: {
                        push: jest.fn()
                    }
                },
                stubs: ['router-link']
            });
        });
        // eslint-disable-next-line require-await
        test('register fonksiyonun çağrılması', async() => {
            const Button = wrapper.find(".register_button");
            Button.trigger("click").then(() => {
                expect(mockFunction).toHaveBeenCalled();
            });
        });
    });


    test('Vue Örneği mi?', () => {
        const wrapper = mount(register, { stubs: ['router-link'] })
        expect(wrapper.vm).toBeTruthy()
    });
    test("Snapshot Testi", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

});