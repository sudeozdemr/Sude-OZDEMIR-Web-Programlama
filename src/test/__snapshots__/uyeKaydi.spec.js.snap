// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uyeOl Snapshot Testi 1`] = `
"<div class=\\"container\\">
  <div class=\\"uyelikHeader\\">
    <div class=\\"logoBox\\"><img src=\\"src/assets/register.png\\" alt=\\"yok\\" style=\\"width: 60px; margin-top: 7px;\\"></div>
    <h2>ÜYE OLMAK İSTİYORUM</h2>
  </div>
  <form>
    <div><label>Adınız</label> <input type=\\"text\\" required=\\"required\\"> <label>Soyadınız</label> <input type=\\"text\\" required=\\"required\\"></div> <label>E-posta</label> <input type=\\"text\\" required=\\"required\\"> <label>Şifre</label> <input type=\\"password\\" required=\\"required\\"> <label>Şifre Tekrar</label> <input type=\\"password\\" required=\\"required\\"> <label>Doğrula</label>
    <div class=\\"reCaptcha\\"><label class=\\"beniHatirla\\"><input type=\\"checkbox\\">Ben robot değilim
      </label> <img src=\\"src/assets/recaptcha.png\\"></div> <label class=\\"beniHatirla\\"><input type=\\"checkbox\\">Özel kampanyalardan ilk sen haberdar ol
      Kampanya ve Bülten, SMS, E-Postalarını Aydınlatma ve Rıza Metni
      kapsamında almak istiyorum</label> <input type=\\"submit\\" value=\\"HESABIMI OLUŞTUR\\" class=\\"register_button\\">
  </form>
</div>"
`;
